llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp

```diff
@@ -530,8 +530,9 @@ bool RISCVInstructionSelector::select(MachineInstr &MI) {
   MachineFunction &MF = *MBB.getParent();
   MachineIRBuilder MIB(MI);
 
-  preISelLower(MI, MIB);
   const unsigned Opc = MI.getOpcode();
+  bool OpcWasGSplatVector = Opc == TargetOpcode::G_SPLAT_VECTOR;
+  preISelLower(MI, MIB);
 
```

Comment:
The Opc needs to be loaded after preISelLower in case the opcode is changed by preISelLower